internal/pkg/deploy/deploy.go

```diff
@@ -16,3 +36,135 @@ type ResourceEvent struct {
 	Status       string
 	StatusReason string
 }
+
+type resourceGetter interface {
+	GetResourcesByTags(resourceType string, tags map[string]string) ([]string, error)
+}
+
+type configStoreClient interface {
+	GetEnvironment(appName string, environmentName string) (*config.Environment, error)
+	ListEnvironments(appName string) ([]*config.Environment, error)
+	GetService(appName, svcName string) (*config.Service, error)
+}
+
+// Store is in charge of fetching the service deployment information.
+type Store struct {
+	rgClient     resourceGetter
+	configStore  configStoreClient
+	initRgClient func(*Store, string, string) error
+}
+
+// NewStore returns a new store.
+func NewStore() (*Store, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("connect to config store: %w", err)
+	}
+	return &Store{
+		configStore: store,
+		initRgClient: func(s *Store, appName, envName string) error {
+			env, err := s.configStore.GetEnvironment(appName, envName)
+			if err != nil {
+				return fmt.Errorf("get environment %s, %w", envName, err)
+			}
+			p := session.NewProvider()
+			sess, err := p.FromRole(env.ManagerRoleARN, env.Region)
+			if err != nil {
+				return fmt.Errorf("create new session from env role: %w", err)
+			}
+			s.rgClient = rg.New(sess)
+			return nil
+		},
+	}, nil
+}
+
+// ListDeployedServices returns all the deployed service in a specific environment of an application.
+func (s *Store) ListDeployedServices(appName string, envName string) ([]*config.Service, error) {
+	err := s.initRgClient(s, appName, envName)
+	if err != nil {
+		return nil, err
+	}
+	svcARNs, err := s.rgClient.GetResourcesByTags(ecsServiceResourceType, map[string]string{
+		AppTagKey: appName,
+		EnvTagKey: envName,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("get resources by Copilot tags: %w", err)
+	}
+	svcs := make([]*config.Service, len(svcARNs))
+	for ind, svcARN := range svcARNs {
+		svcName, err := s.getServiceName(svcARN)
+		if err != nil {
+			return nil, err
+		}
+		svc, err := s.configStore.GetService(appName, svcName)
+		if err != nil {
+			return nil, fmt.Errorf("get service %s: %w", svcName, err)
+		}
+		svcs[ind] = svc
+	}
+	return svcs, nil
+}
+
+// ListEnvironmentsDeployedTo returns all the environment that a service is deployed in.
+func (s *Store) ListEnvironmentsDeployedTo(appName string, svcName string) ([]*config.Environment, error) {
+	var envsWithDeployment []*config.Environment
+	envs, err := s.configStore.ListEnvironments(appName)
+	if err != nil {
+		return nil, fmt.Errorf("list environment for app %s: %w", appName, err)
+	}
+	for _, env := range envs {
+		err := s.initRgClient(s, appName, env.Name)
```

Comment:
This will make an additional call to SSM to re-fetch the environment, although we already retrieved all of them.

What do you think of having two constructors like this in `Store`?
```go
type Store struct {
	configStore  configStoreClient
	newRgClientFromIDs   func(app, env string) (resourceGetter, error)
	newRgClientFromRole  func(roleARN, region string) (resourceGetter, error)
}
```

Reply:
ooh that's a good call!